### ISSUE TITLE
OpenInfraMap petroleum layer erroring with TIMED_OUT

### ISSUE THREAD

**Issue by jywarren** [publiclab/leaflet-environmental-layers#147]
`Failed to load resource: net::ERR_CONNECTION_TIMED_OUT` on:

https://publiclab.github.io/leaflet-environmental-layers/example/#8/30.554/-92.606/BL2-OIMPetroleum

I believe the tiles themselves are not loading: https://tiles-c.openinframap.org/petroleum/5/5/12.png

Maybe we were blocked for hitting their tiles too hard? Or maybe their API changed? 

See #22 #23 for details on the original implementation. Any help appreciated!!!

**Comment by avsingh999** [comment]
@jywarren  Yes, there are also others,  like this please look this https://github.com/publiclab/leaflet-environmental-layers/issues/140 I have opened
thanks : )

**Comment by rkpattnaik780** [comment]
Open Infrastructure Maps have switched from old raster tiles to vector tiles, the raster tiles have been deprecated. They have mentioned about this in a [tweet](https://twitter.com/openinframap/status/1052187991760916480).

The new tileJSON is defined in https://openinframap.org/map.json .

**Comment by rkpattnaik780** [comment]
@jywarren I had mailed about this issue to Mr. Russ Garret, a maintainer of OpenInfraMap. I got this reply : - 

<b>" Those URLs are for the old raster tiles which are no longer in use.
Openinframap now uses Mapbox vector tiles which are less easy to use
in Leaflet unfortunately. You can use the Leaflet.VectorGrid plugin
with the vector tile details in https://openinframap.org/map.json but
you will need to write your own styling. "</b>

I think fixing this issue will require some major changes as well as change in the current structure. What do you say?

**Comment by jywarren** [comment]
Oh, this is really helpful information, though! Thank you!

Hmm, yes, i tihnk we would have to try to refactor this layer
significantly. But perhaps we could start by searching for and linking to
some examples of Mapbox vector tiles being used in Leaflet, as a model to
look to?

**Comment by rkpattnaik780** [comment]
@jywarren  that seems appropriate.

**Comment by SidharthBansal** [comment]
Can we have this in GCI @jywarren @sagarpreet-chadha @ananyaarun ?

**Comment by jywarren** [comment]
Yes, but noting that it will involve some investigating and troubleshooting
to find the problem!

**Comment by SidharthBansal** [comment]
Cool thanks.

Published
